src/Features/Core/Portable/PullMemberUp/QuickAction/AbstractMemberPullerWithQuickAction.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+
+namespace Microsoft.CodeAnalysis.PullMemberUp.QuickAction
+{
+    internal abstract class AbstractMemberPullerWithQuickAction
+    {
+        /// <summary>
+        ///  This method is used to check whether the selected member overrides the member in target.
+        ///  It just checks the members directly declared in the target.
+        /// </summary>
+        protected abstract bool IsDeclarationAlreadyInTarget(INamedTypeSymbol target, ISymbol symbol);
+
+        internal async Task<CodeAction> TryComputeRefactoring(
```

Comment:
these are doing too much work up front. it looks like it does everything up front except code generation. I think you should make everything under this TryComputRefactoring inside code action so that it only runs that when inovked (or previewed)

Reply:
Some of the work here is to check whether we should show up the light bulb. I understand your meaning that we should try to put all slow things into Action. Some of the PR here doesn't follow this idea. I can try to add as much as I can in next commit. (But some checks I think is necessary)